service/prometheus/update.go

```diff
@@ -19,8 +19,8 @@ func UpdateConfig(promcfg config.Config, scrapeConfigs []config.ScrapeConfig) (c
 	}
 
 	// And append the supplied, desired scrape configs.
-	for _, scrapeConfig := range scrapeConfigs {
-		desiredScrapeConfigs = append(desiredScrapeConfigs, &scrapeConfig)
+	for i, _ := range scrapeConfigs {
```

Comment:
This was the issue. Pointers are hard.

Reply:
Everything else is me debugging :D

Reply:
How did it help? Is there some `==` later?

Reply:
Explained in meet.